drivers/interrupt_controller/intc_clic.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+ * Copyright (c) 2025 Andes Technology Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Core Local Interrupt Controller
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/riscv/csr.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/interrupt_controller/riscv_clic.h>
+#include "intc_clic.h"
+
+#if DT_HAS_COMPAT_STATUS_OKAY(riscv_clic)
+#define DT_DRV_COMPAT riscv_clic
+#elif DT_HAS_COMPAT_STATUS_OKAY(nuclei_eclic)
+#define DT_DRV_COMPAT nuclei_eclic
+#endif
+
+struct clic_data {
+	uint8_t nlbits;
+	uint8_t intctlbits;
+};
+
+struct clic_config {
+	mem_addr_t base;
+};
+
+#ifdef CONFIG_PMP_STACK_GUARD
+/*
+ * M-mode CLIC memory-mapped registers are accessible only in M-mode.
+ * Temporarily disable the PMP stack guard (set mstatus.MPRV = 0) to configure
+ * CLIC registers, then restore the PMP stack guard using these macros.
+ */
+#define DISABLE_PMP_STACK_GUARD                                                                    \
+	unsigned int key = irq_lock();                                                             \
```

Comment:
```suggestion
	unsigned int _key = irq_lock();                                                             \
```

To avoid the (unlikely) name clash? (here and for `RESTORE_PMP_STACK_GUARD`)